### DISCUSSION TITLE
MODFLOW executable on Linux

### DISCUSSION THREAD

**Discussion by JanGei** [modflowpy/flopy#1789] [Q&A] [answered]
I recently started working with the Linux operating system (Ubunutu). All of my MODFLOW models are developed under Windows. To solve a given model, the exe_name is specified in the flopy.mf6.modflow.mfsimulation module (https://flopy.readthedocs.io/en/3.3.6/source/flopy.mf6.modflow.mfsimulation.html). Under Windows it is sufficient to add the mf6 executable to the PATH environment in order for the Spyder IDE to find it and use it for model evaluation. 
In Linux however, it is not clear to me, which PATH environment is used by Spyder to look for executables. I added the respective path of the mf6 file to the bashrc file, in PATH and PYTHONPATH and I have edited /etc/environment. However, my Python code is not able to find mf6. Only if I specifiy the exact directory it works. As I obviously want my code to be flexible and device-independent, I want to access the mf6 file via PATH. Any instructions on how to achieve this are greatly appreciated, thank you!

**Comment by dbrakenhoff** [comment, marked as answer]
Hi @JanGei,

It sounds like you've already done this but, just to show you how I've configured it:

I've put the USGS Linux binaries in a folder within my home directory and made sure they are executable:
```bash
(base) david@laptopdavid:~/usgs/bin$ ls -ll
total 144696
-rw-rw-r-- 1 david david     7792 aug  9  2022 code.json
-rwxrwxr-x 1 david david   860848 mrt  8 12:16 crt
-rwxrwxr-x 1 david david  1350760 mrt  8 12:16 gridgen
-rwxrwxr-x 1 david david 11322040 nov 17 16:57 gsflow
-rw-rw-r-- 1 david david 28912392 mrt  8 12:16 libmf6.so
-rwxrwxr-x 1 david david  4433160 mrt  8 12:16 mf2000
-rwxrwxr-x 1 david david  7461296 mrt  8 12:16 mf2005
-rwxrwxr-x 1 david david  7547400 mrt  8 12:16 mf2005dbl
-rwxrwxr-x 1 david david 26949400 mrt  8 12:16 mf6
-rwxrwxr-x 1 david david  6136824 mrt  8 12:16 mflgr
-rwxrwxr-x 1 david david  6190336 mrt  8 12:16 mflgrdbl
-rwxrwxr-x 1 david david  8153120 mrt  8 12:16 mfnwt
-rwxrwxr-x 1 david david  8260408 mrt  8 12:16 mfnwtdbl
-rwxrwxr-x 1 david david  3298416 mrt  8 12:16 mfusg
-rwxrwxr-x 1 david david  3318776 mrt  8 12:16 mfusgdbl
-rwxrwxr-x 1 david david  1356200 mrt  8 12:16 mp6
-rwxrwxr-x 1 david david  1736136 mrt  8 12:16 mp7
-rwxrwxr-x 1 david david  1364640 mrt  8 12:16 mt3dms
-rwxrwxr-x 1 david david  3116632 mrt  8 12:16 mt3dusgs
-rwxrwxr-x 1 david david  3899808 nov 17 16:57 prms
-rwxrwxr-x 1 david david  2478048 mrt  8 12:16 sutra
-rwxrwxr-x 1 david david  5274560 mrt  8 12:16 swtv4
-rwxrwxr-x 1 david david   500120 mrt  8 12:16 triangle
-rwxrwxr-x 1 david david  1072616 mrt  8 12:16 vs2dt
-rwxrwxr-x 1 david david  1435176 mrt  8 12:16 zbud6
-rwxrwxr-x 1 david david   800768 mrt  8 12:16 zonbud3
-rwxrwxr-x 1 david david   875496 mrt  8 12:16 zonbudusg

```

and added the following to my .bashrc file. 

```sh
# >>> add usgs binaries to path >>>
export PATH="$PATH:/home/david/usgs/bin"
# export PYTHONPATH="$PYTHONPATH:/home/david/usgs/bin"
# <<< add usgs binaries to path <<<
```
I recently commented out the PYTHONPATH variable, but it used to be active in my .bashrc file.

Maybe that helps you some?

**Reply by JanGei** [reply]
Thank you so much! I had a mistake in the order of the bashrc file and it works perfectly fine with your definition.

**Reply by wpbonelli** [reply]
Just to expand on @dbrakenhoff's answer, FloPy >=3.3.6 now has a tool that can help install modflow exes without necessarily needing to modify your PATH

https://github.com/modflowpy/flopy/blob/develop/docs/get_modflow.md

Running `get-modflow :` will guide you to select an install location. You may find `get-modflow :python` convenient, as it will install to the current python's bin directory, which should already be on the path assuming you have a conda or `venv` environment activated
